.github/workflows/build.yml

```diff
@@ -0,0 +1,14 @@
+name: Python Linting and Type Checking
+on: [ push, pull_request ]
+jobs:
+  lint_and_type_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Install Just
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://just.systems/install.sh | bash -s -- --to "$HOME/.local/bin"
```

Comment:
I've never heard of this. Could you do the checks without this? I've used pre-commit for black before.

Reply:
Sure. I've never worked with just before either for what it's worth.

Switched the PR to use a shell script in CI, and pre-commit for local development.